turbine/src/broadcast_stage/standard_broadcast_run.rs

```diff
@@ -75,43 +83,41 @@ impl StandardBroadcastRun {
         stats: &mut ProcessShredsStats,
     ) -> Vec<Shred> {
         const SHRED_TICK_REFERENCE_MASK: u8 = ShredFlags::SHRED_TICK_REFERENCE_MASK.bits();
-        let (current_slot, _) = self.current_slot_and_parent.unwrap();
-        match self.unfinished_slot {
-            None => Vec::default(),
-            Some(ref state) if state.slot == current_slot => Vec::default(),
-            Some(ref mut state) => {
-                let reference_tick = max_ticks_in_slot & SHRED_TICK_REFERENCE_MASK;
-                let shredder =
-                    Shredder::new(state.slot, state.parent, reference_tick, self.shred_version)
-                        .unwrap();
-                let (mut shreds, coding_shreds) = shredder.entries_to_shreds(
+        if self.completed {
+            return vec![];
+        }
+        let reference_tick = max_ticks_in_slot & SHRED_TICK_REFERENCE_MASK;
+        let (mut shreds, coding_shreds) =
+            Shredder::new(self.slot, self.parent, reference_tick, self.shred_version)
+                .unwrap()
+                .entries_to_shreds(
                     keypair,
                     &[],  // entries
                     true, // is_last_in_slot,
-                    should_chain_merkle_shreds(state.slot, cluster_type)
-                        .then_some(state.chained_merkle_root),
-                    state.next_shred_index,
-                    state.next_code_index,
+                    should_chain_merkle_shreds(self.slot, cluster_type)
+                        .then_some(self.chained_merkle_root),
+                    self.next_shred_index,
+                    self.next_code_index,
                     true, // merkle_variant
                     &self.reed_solomon_cache,
                     stats,
                 );
-                stats.num_merkle_data_shreds += shreds.len();
-                stats.num_merkle_coding_shreds += coding_shreds.len();
-                self.report_and_reset_stats(true);
-                self.unfinished_slot = None;
-                shreds.extend(coding_shreds);
-                shreds
-            }
+        if let Some(shred) = shreds.iter().max_by_key(|shred| shred.index()) {
```

Comment:
we should always generate 1 shred for the `LAST_SHRED_IN_SLOT` flag right? better to panic here?

Reply:
yeah, but we don't need to rely on that here, and kind of hesitant if the `panic` here is worth it.